# Dataframe for Dynamic Pricing

## Objectives

This document intends to briefly describe one alternative for the ETL process for the Dynamic Pricing problem at Olist.  
As the end product, we would like to have a table with the proper structure for forecasting techniques.  
The datasets will be described along the way.

### Importing packages

In [1]:
import pandas as pd
import datetime

### Dataframe with complete range

Let's start by creating a dataframe with every data point we want to measure. It will contain hourly intervals, equally spaced between two dates.  
You can change the parameters in the cell below:

In [2]:
freq = 'D' # 'H' is for hourly data. You can use 'D' for days, 'W' for last day of week, 'Q' for quarter, etc.

start_date = '1/1/2017' # MM/DD/YYYY
end_date = '4/1/2020' # MM/DD/YYYY

In [3]:
df = pd.DataFrame(data=pd.date_range(start=start_date, end=end_date, freq=freq),
                  columns=['date'])

df.head(2)

date
0 2017-01-01
1 2017-01-02

In [4]:
df.tail(2)

date
1185 2020-03-31
1186 2020-04-01

### Orders dataset

Let's investigate our first dataset "orders.csv". It contains information for every sale that was executed in a given period of time.  
Our main idea is to aggregate sales data and merge it with our dates dataframe (df).

In [5]:
# READING ORDERS DATA AND TRANSFORMING COLUMNS
orders = pd.read_csv('data/orders.csv', dtype={'product_gtin':'str','origin_zip':'str',
                                               'destination_zip':'str'})

orders['purchase_timestamp'] = pd.to_datetime(orders['purchase_timestamp'])

In [6]:
orders.head()

purchase_timestamp                             seller_id   product_gtin  \
0 2019-07-20 11:19:43  a97001a2-a7d2-4bd6-b0d5-1da2a2b8abd9  6952202281890   
1 2019-11-11 20:43:23  a97001a2-a7d2-4bd6-b0d5-1da2a2b8abd9  6951003643197   
2 2020-03-01 16:25:00  c58b83e5-52de-43fe-8399-a7bc312e2aa5  7898907175302   
3 2020-02-04 12:04:10  a97001a2-a7d2-4bd6-b0d5-1da2a2b8abd9  6951003643197   
4 2018-12-14 13:52:19  a97001a2-a7d2-4bd6-b0d5-1da2a2b8abd9  6951003643197   

        product_sku  channel_slug channel_store   price  freight_value  \
0  PRDWRHUYLLMFX2U0  mercadolivre  mercadolivre   82.99          20.86   
1  PRDACYVT895VMO7H  mercadolivre  mercadolivre  119.90           0.00   
2  PRDKTDU50G9FVV57         cnova            ex   39.90          14.24   
3  PRDACYVT895VMO7H  mercadolivre  mercadolivre  130.95           0.00   
4  PRDACYVT895VMO7H  mercadolivre  mercadolivre  128.99          21.51   

   price_freight_shift  commission_product_olist  commission_freight_olist  \
0                  0.0                      0.20                      0.20   
1                  0.0                      0.20                      0.20   
2                  0.0                      0.19                      0.19   
3                  0.0                      0.20                      0.20   
4                  0.0                      0.20                      0.20   

  origin_zip origin_state     origin_city destination_zip destination_state  \
0   01109010           SP       SAO PAULO        89265575                SC   
1   01109010           SP       SAO PAULO        06040470                SP   
2   20910970           RJ  RIO DE JANEIRO        22260210                RJ   
3   01109010           SP       SAO PAULO        89062100                SC   
4   01109010           SP       SAO PAULO        66055000                PA   

  destination_city  
0   Jaraguá do Sul  
1           Osasco  
2   Rio de Janeiro  
3         Blumenau  
4            Belém

In [7]:
orders.product_gtin.nunique()

647

###### Selecting specific product

In the orders dataset, we have sales information for 647 different products.  
For simplification purposes, let's choose one single product for analysis and manipulation of the dataset.  
To do so, choose a single gtin (str) in the cell below.

- gtin = '7908243600137'
- product_sku = 'PRD41RHHBF9NQR74'
- canonical_sku = 'PRDIM5XOLWMUM0HU'


In [8]:
filter_orders = orders[['purchase_timestamp','product_gtin',
                        'price','freight_value',
                        'commission_product_olist','commission_freight_olist']].copy()

# ADD ARTIFICIAL COLUMN WITH THE NUMBER OF ORDERS -> WILL BE USED TO COUNT THE ORDERS BY HOUR
filter_orders['orders'] = 1


# GROUPS ORDERS BY CHOSEN FREQUENCY
filter_orders = filter_orders.groupby([pd.Grouper(key='purchase_timestamp', freq=freq),'product_gtin'])\
                  .agg({'price':'mean', 
                        'freight_value':'mean',
                        'commission_product_olist':'mean',
                        'commission_freight_olist':'mean',
                        'orders':'sum'}).reset_index()

In [9]:
filter_orders.head()

purchase_timestamp   product_gtin  price  freight_value  \
0         2015-09-23  7899893323067  168.0          0.000   
1         2015-10-13  7899893323067  168.0          0.000   
2         2016-04-16  7898152189550  343.9         17.060   
3         2016-06-10  7897889805825  129.9         28.540   
4         2016-06-11  7897889805825  129.9         32.534   

   commission_product_olist  commission_freight_olist  orders  
0                       0.2                       0.0       1  
1                       0.2                       0.0       1  
2                       0.2                       0.0       1  
3                       0.2                       0.0       3  
4                       0.2                       0.0       5

##### merge date_range dataframe and orders

Now that we have the number of sales of a given product by hour, along with its average price, average shipping values and average commissions (percentage), we can join with our date range dataframe.  

The 'NaN' rows are the ones that did not have any sales.

In [10]:
# MERGE DF and Filter orders

merged = pd.merge(df, filter_orders, how='left', left_on='date', right_on='purchase_timestamp')
merged.drop('purchase_timestamp', axis=1, inplace=True)

In [11]:
merged.head()

date product_gtin  price  freight_value  commission_product_olist  \
0 2017-01-01          NaN    NaN            NaN                       NaN   
1 2017-01-02          NaN    NaN            NaN                       NaN   
2 2017-01-03          NaN    NaN            NaN                       NaN   
3 2017-01-04          NaN    NaN            NaN                       NaN   
4 2017-01-05          NaN    NaN            NaN                       NaN   

   commission_freight_olist  orders  
0                       NaN     NaN  
1                       NaN     NaN  
2                       NaN     NaN  
3                       NaN     NaN  
4                       NaN     NaN

In [12]:
# Checking some rows with sales

merged[merged['orders'].notnull()].head()

date   product_gtin   price  freight_value  commission_product_olist  \
5  2017-01-06  7891530258930  579.99          28.55                       0.2   
7  2017-01-08  7898914878487   66.90          18.75                       0.2   
8  2017-01-09  7898914878487   66.90          18.75                       0.2   
10 2017-01-11  7891530258930  579.99          18.23                       0.2   
11 2017-01-11  7897889805825  129.99          29.19                       0.2   

    commission_freight_olist  orders  
5                        0.0     1.0  
7                        0.0     1.0  
8                        0.0     2.0  
10                       0.0     1.0  
11                       0.0     1.0

### Product_info dataset

Another available dataset is the "product_info.csv". We will use it to check when a product was created in our database to fine tune the starting point of our time series.

In [13]:
prod_info = pd.read_csv('data/product_info.csv', dtype={'gtin':'str'})

prod_info['created_at'] = pd.to_datetime(prod_info['created_at'])

In [14]:
filter_prod = prod_info[['created_at','gtin','name']]

In [15]:
filter_prod = filter_prod.groupby(['gtin','name']).min().reset_index()

In [17]:
def truncate_date(x):
    return datetime.date(x.year,x.month,x.day)

In [18]:
filter_prod['created_date'] = filter_prod['created_at'].map(truncate_date)

In [19]:
filter_prod.head()

gtin                                               name  \
0  1001001118649  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
1  1003001113372                       Autoclave Vertical 18 Litros   
2  1003001209488  Mini Centrífuga Para 8 Microtubos Vol. 0,2 A 2...   
3  1234567896618  Veda Fresta De Portão 6,4 Cm Altura Vedapratic...   
4  1286508320126  Diabete composto 90 Cápsulas 500 Mg insulina n...   

           created_at created_date  
0 2019-06-03 14:50:41   2019-06-03  
1 2019-06-21 13:05:50   2019-06-21  
2 2019-12-02 12:38:41   2019-12-02  
3 2019-07-01 09:24:29   2019-07-01  
4 2019-01-24 13:10:00   2019-01-24

In [32]:
prod_info[prod_info['gtin']=='1001001118649']

gtin part_number     canonical_sku seller_product_sku  \
750   1001001118649         NaN  PRDYBAH5LASBS226   PRD8CK2BKF9LDAJ6   
800   1001001118649         NaN  PRD5AOADTTLG8ZDH   PRD8CK2BKF9LDAJ6   
801   1001001118649         NaN  PRD7MNGYFIPKK2CV   PRD8CK2BKF9LDAJ6   
2704  1001001118649         NaN  PRDE1HXHV5LHSGKK   PRD8CK2BKF9LDAJ6   
3016  1001001118649         NaN  PRDVITLYWBJZKIY1   PRD8CK2BKF9LDAJ6   
3098  1001001118649         NaN  PRDVKSOE0WA8PWAB   PRD8CK2BKF9LDAJ6   
4472  1001001118649         NaN  PRD9HKY68HWMFU12   PRD8CK2BKF9LDAJ6   
4481  1001001118649         NaN  PRDK9DSPEV6SOQWL   PRD8CK2BKF9LDAJ6   
4820  1001001118649         NaN  PRD6W7N80XSZ7YOE   PRD8CK2BKF9LDAJ6   
6024  1001001118649         NaN  PRDAVFBEN7OD0SST   PRD8CK2BKF9LDAJ6   
6420  1001001118649         NaN  PRDUBUWYO5JM7CS1   PRD8CK2BKF9LDAJ6   

                                                   name  \
750   Esqueleto 170 cm, Padrão, com Suporte e Base c...   
800   Esqueleto 170 cm, Padrão, com Suporte e Base c...   
801   Esqueleto 170 cm, Padrão, com Suporte e Base c...   
2704  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
3016  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
3098  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
4472  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
4481  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
4820  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
6024  Esqueleto 170 cm, Padrão, com Suporte e Base c...   
6420  Esqueleto 170 cm, Padrão, com Suporte e Base c...   

                                            description    external_id  \
750   Modelo anatômico de esqueleto humano, medindo ...       47565971   
800   Modelo anatômico de esqueleto humano, medindo ...            NaN   
801   Modelo anatômico de esqueleto humano, medindo ...            NaN   
2704  Modelo anatômico de esqueleto humano, medindo ...  MLB1243740907   
3016  Modelo anatômico de esqueleto humano, medindo ...            NaN   
3098  Modelo anatômico de esqueleto humano, medindo ...            NaN   
4472  Modelo anatômico de esqueleto humano, medindo ...       81955387   
4481  Modelo anatômico de esqueleto humano, medindo ...            NaN   
4820  Modelo anatômico de esqueleto humano, medindo ...            NaN   
6024  Modelo anatômico de esqueleto humano, medindo ...            NaN   
6420  Modelo anatômico de esqueleto humano, medindo ...  MLB1243747133   

        channel_slug   price   offer  stock          created_at  \
750            cnova  1260.0  1260.0      9 2019-06-03 14:50:42   
800             zoom  1280.0  1280.0      5 2019-06-03 14:50:45   
801           amazon  1280.0  1280.0      5 2019-06-03 14:50:41   
2704    mercadolivre  1260.0  1260.0      9 2019-06-03 14:50:44   
3016         saraiva  1280.0  1280.0      5 2019-06-03 14:50:44   
3098          luanet  1280.0  1280.0      5 2019-06-03 14:50:43   
4472             b2w  1260.0  1260.0      9 2019-06-03 14:50:41   
4481  madeiramadeira  1280.0  1280.0      5 2019-06-03 14:50:43   
4820       carrefour  1280.0  1280.0      5 2019-06-03 14:50:41   
6024      digitalweb  1280.0  1280.0      5 2019-06-03 14:50:42   
6420    mercadolivre  1260.0  1260.0      9 2019-06-03 14:50:43   

               updated_at  
750   2020-02-12 21:44:30  
800   2019-06-03 14:51:14  
801   2019-08-03 19:07:25  
2704  2020-02-12 21:44:25  
3016  2019-06-03 14:51:02  
3098  2019-06-03 14:51:02  
4472  2020-02-12 22:30:36  
4481  2019-06-03 14:51:14  
4820  2019-12-23 17:07:51  
6024  2019-06-03 14:51:49  
6420  2020-02-12 21:44:24

##### Filter merged based on creation date

In [20]:
merged = pd.merge(merged, filter_prod, left_on='product_gtin', right_on='gtin', how='left')

In [21]:
merged['created_date'] = pd.to_datetime(merged['created_date'])

In [22]:
# Keep dates only where the product existed.
merged = merged[merged['date']>=merged['created_date']].copy()

In [23]:
merged.head()

date   product_gtin   price  freight_value  commission_product_olist  \
5  2017-01-06  7891530258930  579.99          28.55                       0.2   
7  2017-01-08  7898914878487   66.90          18.75                       0.2   
8  2017-01-08  7898914878487   66.90          18.75                       0.2   
10 2017-01-09  7898914878487   66.90          18.75                       0.2   
11 2017-01-09  7898914878487   66.90          18.75                       0.2   

    commission_freight_olist  orders           gtin  \
5                        0.0     1.0  7891530258930   
7                        0.0     1.0  7898914878487   
8                        0.0     1.0  7898914878487   
10                       0.0     2.0  7898914878487   
11                       0.0     2.0  7898914878487   

                                                 name          created_at  \
5                Relógio Technos Ts Carbon Os10er/1r. 2016-07-24 13:30:02   
7   Caixa de Pesca Com 6 Bandejas Articuladas CX-6... 2016-10-02 22:16:36   
8   Caixa de pesca 6 bandejas com divisórias model... 2016-10-17 15:23:04   
10  Caixa de Pesca Com 6 Bandejas Articuladas CX-6... 2016-10-02 22:16:36   
11  Caixa de pesca 6 bandejas com divisórias model... 2016-10-17 15:23:04   

   created_date  
5    2016-07-24  
7    2016-10-02  
8    2016-10-17  
10   2016-10-02  
11   2016-10-17

### Competition

Now, let's get our competitor's prices by the same frequency.

In [24]:
# Read data and format it
competition = pd.read_csv('data/competition.csv', dtype={'gtin':'str'})

competition['updated_at'] = pd.to_datetime(competition['updated_at'])

In [25]:
competition.head()

updated_at           gtin   value                  slug
0 2020-01-29 10:44:18  1001001118649  1260.0   sieve-b2w-submarino
1 2020-01-29 10:43:43  1001001118649  1260.0    sieve-b2w-shoptime
2 2020-01-29 10:42:41  1001001118649  1159.2  sieve-b2w-americanas
3 2020-01-24 15:17:21  1001001118649     0.0    sieve-b2w-shoptime
4 2020-01-24 15:17:20  1001001118649     0.0  sieve-b2w-americanas

In [35]:
# filter out zero values
competition = competition[competition['value'] > 0]

In [43]:
gr = competition.groupby('gtin').agg({'value':['mean', 'std']}).reset_index()

In [44]:
gr.head()

gtin        value           
                         mean        std
0  1001001118649  1256.575000  31.937300
1  1003001113372  5088.323333   4.106978
2  1003001209488  1011.585000  81.363378
3  1234567896618   103.908621   5.497400
4  1286508320126    26.643370   1.758233

In [47]:
comp_m = pd.merge(competition, gr, left_on='gtin', right_on='gtin')

In [50]:
comp_m.columns = ['updated_at', 'gtin', 'value', 'slug', 'mean','std']

In [53]:
comp_m['upper'] = comp_m['mean'].values + 3*comp_m['std'].values
comp_m['lower'] = comp_m['mean'].values - 3*comp_m['std'].values

In [54]:
comp_m.head()

updated_at           gtin   value                  slug      mean  \
0 2020-01-29 10:44:18  1001001118649  1260.0   sieve-b2w-submarino  1256.575   
1 2020-01-29 10:43:43  1001001118649  1260.0    sieve-b2w-shoptime  1256.575   
2 2020-01-29 10:42:41  1001001118649  1159.2  sieve-b2w-americanas  1256.575   
3 2019-11-19 11:23:43  1001001118649  1260.0    olist-mercadolivre  1256.575   
4 2019-11-19 11:02:48  1001001118649  1260.0           olist-cnova  1256.575   

       std      upper      lower  
0  31.9373  1352.3869  1160.7631  
1  31.9373  1352.3869  1160.7631  
2  31.9373  1352.3869  1160.7631  
3  31.9373  1352.3869  1160.7631  
4  31.9373  1352.3869  1160.7631

In [55]:
comp_m.shape

(45919, 8)

In [56]:
# filter comp_m on extremely low and high prices
comp_m = comp_m[(comp_m['value']>= comp_m['lower']) & (comp_m['value']<= comp_m['upper'])]

In [57]:
comp_m.shape

(45412, 8)

In [58]:
# Group by freq and get minimum value
comp_m = comp_m.groupby([pd.Grouper(key='updated_at', freq=freq),'gtin']).min()[['value']].reset_index()

In [59]:
comp_m.head()

updated_at           gtin   value
0 2018-08-22  6941057400297  699.90
1 2018-08-22  7897375164269   12.00
2 2018-08-22  7898066740533  120.90
3 2018-08-22  7898605640461   82.38
4 2018-08-22  7899459304660   39.90

##### merge competition prices with merged dataframe

In [ ]:
#merged['day'] = merged['date'].apply(lambda x: x.date())

In [60]:
merged = pd.merge(merged, comp_m,
         left_on=['date','product_gtin'], right_on=['updated_at', 'gtin'],
         how='left')

merged.rename(columns={'value':'competition_price'}, inplace=True)

merged = merged[['date', 'product_gtin',  'competition_price', 'price', 'freight_value',
       'commission_product_olist', 'commission_freight_olist', 'orders']].copy()

In [61]:
merged.head()

date   product_gtin  competition_price   price  freight_value  \
0 2017-01-06  7891530258930                NaN  579.99          28.55   
1 2017-01-08  7898914878487                NaN   66.90          18.75   
2 2017-01-08  7898914878487                NaN   66.90          18.75   
3 2017-01-09  7898914878487                NaN   66.90          18.75   
4 2017-01-09  7898914878487                NaN   66.90          18.75   

   commission_product_olist  commission_freight_olist  orders  
0                       0.2                       0.0     1.0  
1                       0.2                       0.0     1.0  
2                       0.2                       0.0     1.0  
3                       0.2                       0.0     2.0  
4                       0.2                       0.0     2.0

In [62]:
merged.sort_values(by=['product_gtin', 'date'])

date   product_gtin  competition_price    price  freight_value  \
19707 2020-02-10  1001001118649                NaN  1260.00        104.470   
15532 2019-12-11  1003001113372                NaN  5079.94        263.310   
16953 2020-01-06  1003001209488             939.00   939.00         34.620   
18046 2020-01-20  1003001209488                NaN   939.00         25.210   
9452  2019-08-12  1234567896618                NaN    98.69         22.220   
10457 2019-09-08  1234567896618                NaN    98.69         22.220   
11022 2019-09-22  1234567896618                NaN    98.69         22.220   
11975 2019-10-16  1234567896618                NaN    98.69         26.610   
12811 2019-11-04  1234567896618                NaN    98.69         40.360   
13196 2019-11-11  1234567896618                NaN    98.69         24.415   
13270 2019-11-12  1234567896618                NaN    98.69         16.800   
13343 2019-11-13  1234567896618                NaN    98.69         22.220   
13565 2019-11-17  1234567896618                NaN    98.69         22.220   
13885 2019-11-22  1234567896618                NaN    98.69         22.220   
14152 2019-11-26  1234567896618                NaN    98.69         22.220   
14389 2019-11-29  1234567896618                NaN    98.69         26.610   
14562 2019-11-30  1234567896618                NaN    98.69         22.220   
15234 2019-12-07  1234567896618                NaN    98.69         22.220   
16832 2020-01-04  1234567896618                NaN    98.69         20.220   
17402 2020-01-11  1234567896618                NaN    98.69         26.610   
17522 2020-01-13  1234567896618                NaN    98.69         22.220   
17628 2020-01-14  1234567896618                NaN    98.69         22.220   
18271 2020-01-23  1234567896618                NaN    98.69         22.220   
18454 2020-01-26  1234567896618                NaN    98.69         93.820   
18587 2020-01-28  1234567896618                NaN    98.69         22.680   
18940 2020-02-01  1234567896618                NaN    98.69         22.220   
19203 2020-02-04  1234567896618                NaN    98.69         22.220   
19295 2020-02-05  1234567896618                NaN    98.69         16.800   
19708 2020-02-10  1234567896618                NaN    98.69         19.560   
19993 2020-02-13  1234567896618              98.69    98.69         22.220   
...          ...            ...                ...      ...            ...   
21275 2020-02-29  9788587114617                NaN    33.47         20.675   
21905 2020-03-09  9788587114617                NaN    33.98         44.960   
21997 2020-03-10  9788587114617                NaN    33.98         19.890   
22127 2020-03-12  9788587114617                NaN    33.98         14.000   
16412 2019-12-26  9811000000355                NaN    51.99         21.440   
16831 2020-01-03  9811000000355                NaN    51.99          9.490   
17043 2020-01-06  9811000000355                NaN    51.99          9.490   
17231 2020-01-08  9811000000355                NaN    51.99         17.220   
17318 2020-01-09  9811000000355                NaN    51.99         38.900   
17401 2020-01-10  9811000000355                NaN    51.99         17.875   
17521 2020-01-12  9811000000355                NaN    51.99         20.550   
17703 2020-01-14  9811000000355                NaN    51.99         14.520   
17795 2020-01-15  9811000000355                NaN    51.99         17.220   
18341 2020-01-23  9811000000355                NaN    51.99         17.220   
18586 2020-01-27  9811000000355                NaN    51.99         20.550   
18679 2020-01-28  9811000000355                NaN    51.99         49.710   
18863 2020-01-30  9811000000355                NaN    51.99          9.490   
19016 2020-02-01  9811000000355                NaN    51.99          9.490   
19095 2020-02-02  9811000000355                NaN    51.99          9.490   
19202 2020-02-03  9811000000355   

We will forward fill the competition price.

In [ ]:
merged['competition_price'].fillna(method='ffill', inplace=True)

In [ ]:
merged.head()

In [ ]:
merged.tail()

### Product type and group

Let's gather some information about the type of the analyzed product.

This particular gtin belongs to the product_type "Kit para cabelo" (Hair Care Kit).  
The product belongs to the broader group "beleza_saude" (beauty_health)

In [ ]:
product_type = pd.read_csv('data/product_type.csv', dtype={'value':str})

product_type.head()

product_type[product_type['value']==gtin]

merged.columns

merged = pd.merge(left=merged, right=product_type,
         left_on='product_gtin', right_on='value')[['date', 'product_gtin', 'competition_price',
                                                    'price', 'freight_value', 'commission_product_olist', 
                                                    'commission_freight_olist', 'orders','product_type', 
                                                    'product_type_group']]

In [ ]:
merged.head()

### Olist history of prices
### TODO

For this particular case, we will only read historical data for the selected gtin. This will be changed once olist can share data on other products.  
Due to internal issues, the price history could not be gathered.


In [ ]:
# history = pd.read_csv('data/product_history.csv', dtype={'gtin':str})
# history['updated_at'] = pd.to_datetime(history['updated_at'])

### SOME CHARTS

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns
import chart_studio.plotly as py
import plotly.graph_objs as go
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot

init_notebook_mode(connected=True)

In [ ]:
merged.columns

In [ ]:
trace0 = go.Scatter(x=merged.date,
                   y=merged.orders, mode='lines')

trace1 = go.Scatter(x=merged.date,
                    y=merged.competition_price, mode='markers')

trace2 = go.Scatter(x=merged.date,
                    y=merged.price,
                    mode='markers')

trace3 = go.Scatter(x=merged.date,
                    y=merged.freight_value,
                    mode='lines')


iplot([trace0, trace1, trace2, trace3])

In [ ]:
plt.figure(figsize=(12,8))

plt.style.use('fivethirtyeight')

sns.lineplot(x='date', y='orders', data=merged)
sns.scatterplot(x='date', y='competition_price', data=merged, label='competitor')
sns.scatterplot(x='date', y='price', data=merged, label='olist')

plt.legend()

plt.tight_layout()
plt.show()

In [ ]:
sum(merged['orders'])